aeron-tools/src/main/java/uk/co/real_logic/aeron/tools/MessageStream.java

```diff
@@ -0,0 +1,462 @@
+     * subscriber-side MessageStream).
+     * @return the number of messages that have been either generated
+     * from or inserted into the MessageStream
+     */
+    public long getMessageCount()
+    {
+        return messageCount;
+    }
+
+    /**
+     * Gets the current sequence number of the MessageStream; for a
+     * publisher-side MessageStream, this is the sequence number of
+     * the last message that was generated.  For a subscriber-side
+     * MessageStream, this is the sequence number of the last message
+     * inserted.  If no messages have yet been generated or inserted,
+     * this will return -1.
+     * @return the MessageStream's current sequence number, or -1 if no
+     * messages have been generated or inserted
+     */
+    public long getSequenceNumber()
+    {
+        return sequenceNumber;
+    }
+
+    /** Returns true if the buffer is _probably_ a verifiable message, false otherwise.
+     * This method just looks for a magic word at the beginning of the message; random
+     * data might happen to reproduce one of the magic words about 1 in 2 billion
+     * times.
+     * @param buffer Buffer with a message that may or may not be a verifiable message
+     * @param offset Offset within the buffer where the message starts
+     * @return true if the message appears to be a verifiable message, false otherwise
+     */
+    public static boolean isVerifiable(final DirectBuffer buffer, final int offset)
+    {
+        if ((buffer.capacity() - offset) < HEADER_LENGTH)
+        {
+            return false;
+        }
+        final int magic = buffer.getInt(offset);
+        if ((magic == MAGIC) || (magic == MAGIC_END))
+        {
+            return true;
+        }
+        return false;
+    }
+
+    static void printHex(final DirectBuffer buffer, final int length)
+    {
+        printHex(buffer, 0, length);
+    }
+
+    static void printHex(final UnsafeBuffer buffer, final int length)
+    {
+        printHex(buffer, 0, length);
+    }
+
+    static void printHex(final UnsafeBuffer buffer, final int offset, final int length)
+    {
+        int pos = 0;
+        for (int i = offset; i < (offset + length); i++)
+        {
+            System.out.printf("%02x ", buffer.getByte(i));
+            if (++pos % HEX_PRINT_WIDTH == 0)
+            {
+                System.out.println();
+            }
+        }
+        System.out.println();
+    }
+
+    public static void printHex(final DirectBuffer buffer, final int offset, final int length)
+    {
+        int pos = 0;
+        for (int i = offset; i < (offset + length); i++)
+        {
+            System.out.printf("%02x ", buffer.getByte(i));
+            if (++pos % HEX_PRINT_WIDTH == 0)
+            {
+                System.out.println();
+            }
+        }
+        System.out.println();
+    }
+
+    /** Generates a message of random (within the constraints the MessageStream was
```

Comment:
For one liners it is OK. If 2 or more then top and tail. IntelliJ will do this for you.